File: backend/src/main/scala/microservices/train/objects/SearchTrainsPlannerRequest.scala
package com.typesafe.travel.train.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SearchTrainsPlannerRequest(fromStation: Option[String], toStation: Option[String], date: Option[String])
object SearchTrainsPlannerRequest:
  given sourceEncoder: Encoder[SearchTrainsPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[SearchTrainsPlannerRequest] = deriveDecoder

File: frontend/src/microservices/train/objects/SearchTrainsPlannerRequest.ts
// 本文件定义 train 模块的 `SearchTrainsPlannerRequest`，用于按出发地、到达地和日期搜索火车并提供 JSON 编解码。

export type SearchTrainsPlannerRequest = {
  fromStation?: string
  toStation?: string
  date?: string
}
export const searchTrainsPlannerRequestFromJson = (json: string): SearchTrainsPlannerRequest =>
  JSON.parse(json) as SearchTrainsPlannerRequest

export const searchTrainsPlannerRequestToJson = (value: SearchTrainsPlannerRequest): string =>
  JSON.stringify(value)
